runtime/jvmti/jvmtiStackFrame.cpp

```diff
@@ -798,6 +799,32 @@ popFrameCheckIterator(J9VMThread * currentThread, J9StackWalkState * walkState)
 	return J9_STACKWALK_KEEP_ITERATING;
 }
 
+static UDATA
+jvmtiGetFrameLocationIterator(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+#if JAVA_SPEC_VERSION >= 21
+	J9Method *method = walkState->method;
+	/* walkState->method can never be NULL since the J9_STACKWALK_VISIBLE_ONLY flag is set. */
+	Assert_JVMTI_true(NULL != method);
+
+	J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+	U_32 extendedModifiers = getExtendedModifiersDataFromROMMethod(romMethod);
+
+	if (J9_ARE_ANY_BITS_SET(extendedModifiers, CFR_METHOD_EXT_JVMTIMOUNTTRANSITION_ANNOTATION)) {
+		walkState->framesWalked -= 1;
```

Comment:
What is the reason for modifying the stack walker field here and below?

Reply:
Because later after the walk, `framesWalked` will be checked and if it's zero `JVMTI_ERROR_NO_MORE_FRAMES` will be returned.

Reply:
To be clear, can you just use a true/false value in one of the fields to determine if the required frame has been reached? You would then only need to not decrement the depth when hidden frames are detected.